app/frontend/src/features/messages/groupchats/groupChatsSlice.ts

```diff
@@ -0,0 +1,123 @@
+import { flow, getParent, Instance, types } from "mobx-state-tree";
+import { GroupChat, Message } from "../../../pb/conversations_pb";
+import { service } from "../../../service";
+
+export const initialState = {
+  groupChats: [],
+  error: "",
+  loading: false,
+  groupChatView: {
+    groupChat: null,
+    messages: [],
+    error: "",
+    loading: false,
+  },
+};
+
+export const GroupChatsModel = types
+  .model({
+    groupChats: types.array(types.frozen<GroupChat.AsObject>()),
```

Comment:
I can see why JS devs who don't want to make the full switch to TypeScript would love this, but since we are using TS, this feels less fluent to me.

Reply:
The main reason for this is runtime type-checking I think. But the designtime type checking also works. It is not for not switching to ts, but to structure your model layer in an object oriented way. So methods are bound to objects. The view layer is functional.